### PR TITLE
Added /publish/all API doc

### PR DESCRIPTION
**Pull request by avasquez614**
Ticket craftercms/craftercms#5591

### CODE REVIEW

File: src/main/api/2/studio-api-2.yaml [hunk 1]
```diff
@@ -3807,6 +3807,48 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
+  /publish/all:
+    post:
+      tags:
+        - publishing
+      summary: Publishes all changes between the sandbox and the publishing environment (staging or live)
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Publishes the entire project to the target (staging or live)

File: src/main/api/2/studio-api-2.yaml [hunk 2]
```diff
@@ -3807,6 +3807,48 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
+  /publish/all:
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml:3810]: Must update the version number of this API spec too.

File: src/main/api/2/studio-api-2.yaml [hunk 3]
```diff
@@ -3807,6 +3807,48 @@ paths:
...
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                siteId:
+                  type: string
+                  description: site ID
+                environment:
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: publishingTarget (we no longer use the word environment).

File: src/main/api/2/studio-api-2.yaml [hunk 4]
```diff
@@ -3807,6 +3807,48 @@ paths:
...
+                    type: string
+                    description: the publishing environment (staging or live). If not specified, live is assumed
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: the publishing target ...

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/api/2/studio-api-2.yaml (hunk 1 above)
Comment (avasquez614) [reply src/main/api/2/studio-api-2.yaml]: Done

File: src/main/api/2/studio-api-2.yaml (hunk 2 above)
Comment (avasquez614) [reply src/main/api/2/studio-api-2.yaml:3810]: Done

File: src/main/api/2/studio-api-2.yaml (hunk 3 above)
Comment (avasquez614) [reply src/main/api/2/studio-api-2.yaml]: Done

File: src/main/api/2/studio-api-2.yaml (hunk 4 above)
Comment (avasquez614) [reply src/main/api/2/studio-api-2.yaml]: Done

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**